### PR TITLE
CrowdStrike Streaming v2 - init client outside the discover/refresh stream loop

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] Ready

fixes https://github.com/demisto/etc/issues/32265

## Description
1. removed the discovery/refresh task, instead doing:
    - discovering stream in the beginning of the fetch event loop
    - refreshing the stream every 25 minutes in the loop
2. added container id to some of the log prints
3. added enforcement of min 1 minute for token expiry time

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

**Review by anara123 — APPROVED** [review]

File: Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py [hunk 1]
```diff
@@ -15,14 +15,14 @@
...
-MINUTES_25 = 25 * 60
 MINUTES_30 = 30 * 60
 TIME_BUFFER_1_MINUTE = 1 * 60
 OK_STATUS_CODE = 200
 CREATED_STATUS_CODE = 201
 UNAUTHORIZED_STATUS_CODE = 401
 TOO_MANY_REQUESTS_STATUS_CODE = 429
-NOT_FOUND_STATUS_CODE = 404
+
+CONTAINER_ID = os.environ.get('HOSTNAME')
```
Comment (glicht) [inline Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py:25]: Nice

File: Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py [hunk 2]
```diff
@@ -109,11 +108,9 @@ async def _http_request(self, method, url_suffix, full_url=None, headers=None, a
                     )
                     time.sleep(time_to_wait)
```
Comment (glicht) [inline Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py:109]: Shouldn't we be using the asycio sleep version as we are in a async function?

File: Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py [hunk 3]
```diff
@@ -238,78 +223,76 @@ async def fetch_event(
...
-                                    last_fetch_stats_print = datetime.utcnow()
-                    except Exception as e:
-                        demisto.debug(f'Failed to fetch event: {e} - Going to sleep for 10 seconds and then retry -'
-                                      f' {traceback.format_exc()}')
-                        await sleep(10)
+            demisto.debug(f'Starting event fetch loop. Container ID: {CONTAINER_ID}')
+            self.client = Client(
+                base_url=self.base_url, app_id=self.app_id, verify_ssl=self.verify_ssl, proxy=self.proxy
+            )
+            await self._discover_stream()
```
Comment (glicht) [inline Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py:230]: This was previously inside a try/catch. Are we sure we want it outside of the try/catch?

File: Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py (hunk 2 above)
Comment (Itay4) [reply Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py]: yea, changed

File: Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py (hunk 3 above)
Comment (Itay4) [reply Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py]: pushed the whole code in the loop into the try-except

**Review by glicht — APPROVED** [review]
Looks good. Let's merge

**Pull request merged**